src/corelibs/U2View/src/ov_sequence/ADVGraphModel.cpp

```diff
@@ -19,175 +19,116 @@
-            float v1 = data[startIdx];
-            float v2 = data[startIdx + 1];
-            float k = v2 - v1;
-            float valInStart = v2 - k * startDiff;
-            part1 = startDiff * (valInStart + v2) / 2;
-        }
-        int firstIdxInRange = int(ceil(start));
-        // calculating part2
-        for (int i = firstIdxInRange; i < endIdx; i++) {
-            part2 += data[i];
-        }
-        // calculating part3
-        if (endDiff > ACCEPTABLE_FLOAT_PRESISION_LOSS && endIdx + 1 < (int)data.size()) {
-            float v1 = data[endIdx];
-            float v2 = data[endIdx + 1];
-            float k = v2 - v1;
-            float valInEnd = v1 + k * endDiff;
-            part3 = endDiff * (v1 + valInEnd) / 2;
-        }
-        // sum
-        result = (part1 + part2 + part3) / range;
-    } else {
-        // result constructed from 1 part: ave[start, end], no data points between
-        int startIdx = int(start);
-        float startDiff = start - float(startIdx);
-        float endDiff = startDiff + range;
-        assert(endDiff < 1);
-        float v1 = data[startIdx];
-        float v2 = data[startIdx + 1];
-        float k = v2 - v1;
-        float valInStart = v1 + k * startDiff;
-        float valInEnd = v1 + k * endDiff;
-        result = (valInEnd + valInStart) / 2;
-    }
-    return result;
+float GSequenceGraphUtils::getMinValue(float value1, float value2) {
+    return isUndefined(value1) ? value2 : isUndefined(value2) ? value1
+                                                              : qMin(value1, value2);
 }
 
-void GSequenceGraphUtils::fitToScreen(const QVector<float> &data, int dataStartBase, int dataEndBase, QVector<float> &results, int resultStartBase, int resultEndBase, int screenWidth, float unknownVal) {
-    // BUG:422: use intervals and max/min values instead of average!
-    float basesPerPixel = (resultEndBase - resultStartBase) / (float)screenWidth;
-    float basesInDataPerIndex = (dataEndBase - dataStartBase) / (float)(data.size() - 1);
-    float currentBase = resultStartBase;
-    results.reserve(results.size() + screenWidth);
-    for (int i = 0; i < screenWidth; i++, currentBase += basesPerPixel) {
-        float dataStartIdx = (currentBase - basesPerPixel / 2 - dataStartBase) / basesInDataPerIndex;
-        float dataEndIdx = (currentBase + basesPerPixel / 2 - dataStartBase) / basesInDataPerIndex;
-        dataStartIdx = qMax((float)0, dataStartIdx);
-        dataEndIdx = qMin((float)data.size() - 1, dataEndIdx);
-        float nDataPointsToAverage = dataEndIdx - dataStartIdx;
-        float val = unknownVal;
-        if (nDataPointsToAverage >= ACCEPTABLE_FLOAT_PRESISION_LOSS) {
-            val = calculateAverage(data, dataStartIdx, nDataPointsToAverage);
-        }
-        results.append(val);
+float GSequenceGraphUtils::getMaxValue(float value1, float value2) {
+    return isUndefined(value1) ? value2 : isUndefined(value2) ? value1
```

Comment:
What if both of them are undefined?

Reply:
The result will be undefined. Undefined values in viewPoints and visibleMin/Max must be checked and handled correctly.